plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkProvider.java

```diff
@@ -26,80 +26,33 @@
  */
 public class CloudSdkProvider extends ContextFunction {
 
-  private final IPreferenceStore preferences;
-
-   public CloudSdkProvider(IPreferenceStore preferences) {
-     // todo drop the null check and use the no args constructor instead
-     if (preferences == null) {
-       preferences = PreferenceInitializer.getPreferenceStore();
-     }
-     this.preferences = preferences;
-   }
-   
-   public CloudSdkProvider() {
-     this(PreferenceInitializer.getPreferenceStore());
-   }
+  private final IPreferenceStore preferences = PreferenceInitializer.getPreferenceStore();
 
   /**
    * Return the {@link CloudSdk} instance from the configured or discovered Cloud SDK.
    * 
    * @return the configured {@link CloudSdk} or {@code null} if no SDK could be found
    */
   public CloudSdk getCloudSdk() {
-    return createBuilder(null).build();
-  }
-
-  /**
-   * Return the location of the configured or discovered Cloud SDK.
-   * 
-   * @return the configured location or {@code null} if the SDK could not be found
-   */
-  public File getCloudSdkLocation() {
-    return resolveSdkLocation();
-  }
-
-  /**
-   * Return a {@link CloudSdk.Builder} instance, suitable for creating a {@link CloudSdk} instance.
-   * {@code location}, if not null, is used as the location to the Cloud SDK, otherwise the
-   * configured or discovered Cloud SDK will be used instead.
-   * 
-   * @param location if not {@code null}, overrides the default location for the SDK; can be a
-   *        {@linkplain String}, {@linkplain File}, or {@linkplain Path}.
-   * @return a builder, or {@code null} if the Google Cloud SDK cannot be located
-   */
-  public CloudSdk.Builder createBuilder(File location) {
-    // perhaps should try to be cleverer in case location references the .../bin/gcloud
-    if (location == null || !location.exists()) {
-      location = resolveSdkLocation();
-    }
-    if (location == null || !location.exists()) {
-      return null;
-    }
-    return new CloudSdk.Builder().sdkPath(location);
-  }
-
-  /**
-   * Attempt to resolve the Google Cloud SDK from the configured location or try to discover its
-   * location.
-   * 
-   * @return the location, or {@code null} if not found
-   */
-  private File resolveSdkLocation() {
-    String value = preferences.getString(PreferenceConstants.CLOUDSDK_PATH);
-    if (value != null && !value.isEmpty()) {
-      return new File(value);
-    }
-    Path discovered = PathResolver.INSTANCE.getCloudSdkPath();
-    if (discovered != null) {
-      return discovered.toFile();
+    CloudSdk.Builder sdkBuilder = new CloudSdk.Builder();
+    
+    String configuredPath = preferences.getString(PreferenceConstants.CLOUDSDK_PATH);
+    
+    // Let's use the configured path, if there is one. Otherwise, the lib will try to discover the path.
+    if (configuredPath != null && !configuredPath.isEmpty()) {
+    	// TODO(joaomartins): What happens when the provided path is invalid? Tools lib isn't
```

Comment:
@briandealwis this is (or should be) all done in the library. I don't like the pattern where the library is surfacing a resolver and want to turn that off. Even in my original implementation of the commlib, I just used the PathResolver because we were using some legacy code from the GAE SDK to stage.

It should be as simple as Eclipse requesting a CloudSdk instance from the library and the library returning a valid one, or error. Let me know if you see any issues with this.